collector/cpu_linux.go

```diff
@@ -28,28 +29,58 @@ const (
 	userHz = 100
 )
 
-type statCollector struct {
-	cpu          *prometheus.Desc
-	intr         *prometheus.Desc
-	ctxt         *prometheus.Desc
-	forks        *prometheus.Desc
-	btime        *prometheus.Desc
-	procsRunning *prometheus.Desc
-	procsBlocked *prometheus.Desc
+type cpuCollector struct {
+	cpu                *prometheus.Desc
+	cpuFreq            *prometheus.Desc
+	cpuFreqMin         *prometheus.Desc
+	cpuFreqMax         *prometheus.Desc
+	cpuCoreThrottle    *prometheus.Desc
+	cpuPackageThrottle *prometheus.Desc
+	intr               *prometheus.Desc
+	ctxt               *prometheus.Desc
+	forks              *prometheus.Desc
+	btime              *prometheus.Desc
+	procsRunning       *prometheus.Desc
+	procsBlocked       *prometheus.Desc
 }
 
 func init() {
-	Factories["stat"] = NewStatCollector
+	Factories["cpu"] = NewCpuCollector
 }
 
-// NewStatCollector returns a new Collector exposing kernel/system statistics.
-func NewStatCollector() (Collector, error) {
-	return &statCollector{
+// NewCpuCollector returns a new Collector exposing kernel/system statistics.
+func NewCpuCollector() (Collector, error) {
+	return &cpuCollector{
 		cpu: prometheus.NewDesc(
 			prometheus.BuildFQName(Namespace, "", "cpu"),
```

Comment:
Hm, a metric which is the prefix of other metrics is a first and could be confusing. Changing that is outside of the scope of this PR, but we should finally decide on a timeline to rename all these metrics.